### PR TITLE
Add initial implementation

### PR DESCRIPTION
**Pull request by niksy**
As per https://github.com/sublimelsp/repository/pull/124, this PR adds initial implementation for language server.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# LSP-twiggy
+
+Twig support for Sublime’s LSP.
+
+Provided through [twiggy-language-server](https://github.com/moetelo/twiggy/tree/master/packages/language-server).
+
+### Installation
+
+* Install [LSP](https://packagecontrol.io/packages/LSP) and `LSP-twiggy` via Package Control.
```
Comment (predragnikolic) [inline README.md:9]: Add a step to install the Twig Syntax here -> https://packagecontrol.io/packages/Twig.

Comment (niksy) [reply README.md:9]: Done!

**Comment by rchl** [comment]
I don't know how important that functionality is but I see that the VSCode extension has client-side logic to request (additional?) completions from the server for the virtual `embedded-content` schema: https://github.com/moetelo/twiggy/blob/1e9acaeb47e8887a69181ebe5cc418392afba3be/packages/vscode/src/extension.ts#L136-L143
Looks like completions for the HTML parts of the template or something.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+* Install [Twig syntax higlight package](https://packagecontrol.io/packages/Twig).
```
Comment (predragnikolic) [inline README.md:10]:
```suggestion
* Install [LSP](https://packagecontrol.io/packages/LSP), [Twig Syntax](https://packagecontrol.io/packages/Twig) and [LSP-twiggy](https://packagecontrol.io/packages/LSP-twiggy) via Package Control.
```

**Comment by niksy** [comment]
> I don't know how important that functionality is but I see that the VSCode extension has client-side logic to request (additional?) completions from the server for the virtual `embedded-content` schema: [moetelo/twiggy@`1e9acae`/packages/vscode/src/extension.ts#L136-L143](https://github.com/moetelo/twiggy/blob/1e9acaeb47e8887a69181ebe5cc418392afba3be/packages/vscode/src/extension.ts#L136-L143) Looks like completions for the HTML parts of the template or something.

Haven’t seen any issues with this, all other buggy stuff has already been submitted as issue by me or someone else, so maybe this will also be addressed as part of making language server more non-VS Code friendly.

**Comment by rchl** [comment]
I've submitted https://github.com/moetelo/twiggy/issues/54 btw (unrelated to my previous comment).

To make file watching work for now, you can manually register it - see https://github.com/sublimelsp/LSP-eslint/blob/356d5b3d0ff86647481f9bebf37185804e5ca1d4/LSP-eslint.sublime-settings#L128-L134

**Comment by moetelo** [comment]
> this functionality just makes sure that HTML-specific completions are provided when outside of twig templates.

That's correct. The purpose of the entire `provideCompletionItem` middleware is to query the Twiggy LSP to check whether we're in an HTML or Twig context. If we're in HTML, it then requests additional HTML completions from VS Code's HTML LSP.

----

By "HTML context," I mean:

`some-file.twig`
```twig
<div> | </div>
```

By "Twig context," I mean:

`some-file.twig`
```twig
<div>{{ | }}</div>
```

**Comment by rchl** [comment]
I see. That's very VSCode-specific approach. Ideally the server would bundle the html service and utilize that directly. But oh well, it doesn't seem essential.

(In theory one can run both LSP-html and LSP-twiggy at the same time but I suppose that might create unnecessary completions within context of interpolations)

**Comment by rchl** [comment]
> I've submitted [moetelo/twiggy#54](https://github.com/moetelo/twiggy/issues/54) btw (unrelated to my previous comment).
> 
> To make file watching work for now, you can manually register it - see [sublimelsp/LSP-eslint@`356d5b3`/LSP-eslint.sublime-settings#L128-L134](https://github.com/sublimelsp/LSP-eslint/blob/356d5b3d0ff86647481f9bebf37185804e5ca1d4/LSP-eslint.sublime-settings#L128-L134)

@niksy can you make that change in the mean time?

**Comment by niksy** [comment]
@rchl done!

**Review by rchl — APPROVED** [review]

**Pull request merged**